heudiconv/parser.py

```diff
@@ -278,15 +278,9 @@ def infotoids(
             )
             lgr.info("Study session for %r", study_session_info)
 
-            if study_session_info in study_sessions:
-                if grouping != "all":
-                    # MG - should this blow up to mimic -d invocation?
-                    lgr.warning(
-                        "Existing study session with the same values (%r)."
-                        " Skipping DICOMS %s",
-                        study_session_info,
-                        seqinfo.values(),
-                    )
-                    continue
+            if grouping != "all":
+                assert (study_session_info not in study_sessions), (
+                    f"Existing study session {study_session_info} "
+                    f"already in analyzed sessions {study_sessions.keys()}")
```

Comment:
I have checked 1366 heudiconv.log files from conversions and found none with "Existing study session with the same values". So I think at least I should be safe ;) let's see how error treats us.